TestProject/Lesson15.cpp

```diff
@@ -41,9 +41,17 @@ namespace Lesson15
 		return acc;
 	}
 	//PS we think that students ALWAYS sorted, so we return student [0], otherwise we need to loop throw it
-	Student* BestStudent(Student students[])
+	Student* BestStudent(Student students[], int size)
 	{
-		return &students[0];
+		int bestStudentId = 0;
+		for (int i = 1; i < size; i++)
+		{
+			if (students[i].Avarage() > students[bestStudentId].Avarage())
```

Comment:
you shouldn't calculate average for best student on each loop iteration. you can calculate it only when update it